src/test/java/com/google/devtools/build/lib/bazel/repository/posix/LocalConfigPlatformFunctionPosixTest.java

```diff
@@ -0,0 +1,83 @@
+// Copyright 2018 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.lib.bazel.repository;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import com.google.devtools.build.lib.analysis.ConfiguredTarget;
+import com.google.devtools.build.lib.analysis.platform.ConstraintSettingInfo;
+import com.google.devtools.build.lib.analysis.platform.ConstraintValueInfo;
+import com.google.devtools.build.lib.analysis.platform.PlatformInfo;
+import com.google.devtools.build.lib.analysis.platform.PlatformProviderUtils;
+import com.google.devtools.build.lib.analysis.util.BuildViewTestCase;
+import com.google.devtools.build.lib.cmdline.Label;
+import com.google.devtools.build.lib.util.CPU;
+import com.google.devtools.build.lib.util.OS;
+import java.util.Collection;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+/** Posix-only tests for {@link LocalConfigPlatformFunction}. */
+@RunWith(Enclosed.class)
+public class LocalConfigPlatformFunctionPosixTest {
+
+  /** Tests on overall functionality. */
+  @RunWith(JUnit4.class)
+  public static class FunctionTest extends BuildViewTestCase {
+    private static final ConstraintSettingInfo CPU_CONSTRAINT =
+        ConstraintSettingInfo.create(Label.parseAbsoluteUnchecked("@bazel_tools//platforms:cpu"));
+    private static final ConstraintSettingInfo OS_CONSTRAINT =
+        ConstraintSettingInfo.create(Label.parseAbsoluteUnchecked("@bazel_tools//platforms:os"));
+
+    private static final ConstraintValueInfo X86_64_CONSTRAINT =
+        ConstraintValueInfo.create(
+            CPU_CONSTRAINT, Label.parseAbsoluteUnchecked("@bazel_tools//platforms:x86_64"));
+    private static final ConstraintValueInfo LINUX_CONSTRAINT =
+        ConstraintValueInfo.create(
+            OS_CONSTRAINT, Label.parseAbsoluteUnchecked("@bazel_tools//platforms:linux"));
+
+    @Test
+    public void generateConfigRepository() throws Exception {
+      CPU.setForTesting(CPU.X86_64);
+      OS.setForTesting(OS.LINUX);
+
+      //    rewriteWorkspace("local_config_platform(name='local_config_platform_test')");
```

Comment:
I have an in-progress patch that removes this line. Can you go ahead and remove it now?

Reply:
Done.